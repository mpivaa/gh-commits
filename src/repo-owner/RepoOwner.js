// @flow
import * as React from 'react';
import styled from 'styled-components';
import { find } from 'lodash';
import { spacing, color, radius } from '../theme';
import { Route } from 'react-router-dom';
import { Heading, Avatar } from '../ui';
import { RepoList } from './RepoList';
import { Repo } from '../repo';
import type { Repository } from '../repo';
import type { Owner } from './types';

let RepoOwnerStyle = styled.div`
  height: 100%;
  display: grid;
  grid-template-columns: 300px 1fr;
  border: 1px solid ${color('lightgrey')};
  border-radius: ${radius};
  box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.02);
  overflow: hidden;
`;

let Sidebar = styled.div`
  display: flex;
  flex-direction: column;
  border-right: 1px solid ${color('lightgrey')};
  overflow: hidden;
`;

let Header = styled.div`
  flex: 0;
  display: grid;
  grid-template-columns: 32px 1fr;
  grid-gap: ${spacing('sm')};
  align-items: center;
  padding: ${spacing('md')};
  border-bottom: 1px solid ${color('lightgrey')};
`;

type Props = {
  owner: Owner,
  repos: Repository[]
};

function RepoOwner({ owner, repos }: Props) {
  return (
    <RepoOwnerStyle>
      <Sidebar>
        <Header>
          <Avatar src={owner.avatarUrl} />
          <Heading as="h3">{owner.login}</Heading>
        </Header>
        <RepoList repos={repos} owner={owner} />
      </Sidebar>
      <Route
        exact
        path="/:owner/:repo"
        render={({
          match: {
            params: { repo }
          }
        }) => <Repo repo={find(repos, { name: repo })} />}
      />
    </RepoOwnerStyle>
  );
}

export { RepoOwner };
