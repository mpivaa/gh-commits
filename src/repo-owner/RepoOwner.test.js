import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import { RepoOwner } from './RepoOwner';

it('renders a list of repositories', () => {
  let tree = renderer
    .create(
      <StaticRouter location="/somelogin" context={{}}>
        <RepoOwner
          owner={{
            avatarUrl: '...',
            login: 'somelogin'
          }}
          repos={[
            {
              name: 'somerepo',
              owner: {
                avatarUrl: '...',
                login: 'somelogin'
              }
            }
          ]}
        />
      </StaticRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
