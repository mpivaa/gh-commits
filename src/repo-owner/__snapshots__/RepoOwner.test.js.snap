// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of repositories 1`] = `
<div
  className="sc-kGXeez hDqTAq"
>
  <div
    className="sc-kpOJdX ljRZrF"
  >
    <div
      className="sc-dxgOiQ ewscA"
    >
      <img
        alt="Avatar"
        className="sc-EHOje cFNCFP"
        src="..."
      />
      <h3
        className="sc-bdVaJa hHOWiG"
      >
        somelogin
      </h3>
    </div>
    <ul
      className="sc-gzVnrw bLQPWQ"
    >
      <a
        className="sc-gZMcBi cqlAIU"
        href="/somelogin/somerepo"
        onClick={[Function]}
      >
        <li
          className="sc-htoDjs eJHaKy"
        >
          <div
            className="sc-dnqmqq kxWGqn"
          >
            somerepo
          </div>
          <p
            className="sc-iwsKbI dUJVnR"
          />
        </li>
      </a>
    </ul>
  </div>
</div>
`;
